strawberry/chalice/views.py

```diff
@@ -0,0 +1,153 @@
+import functools
+from http import HTTPStatus
+
+from chalice.app import CaseInsensitiveMapping, Request, Response
+from strawberry.chalice.graphiql import render_graphiql_page
+from strawberry.http import GraphQLHTTPResponse, process_result
+from strawberry.schema import BaseSchema
+from strawberry.types import ExecutionResult
+
+
+class GraphQLView:
+    def __init__(self, schema: BaseSchema, render_graphiql=True):
+        self._schema: BaseSchema = schema
+        self.graphiql: bool = render_graphiql
+
+    @staticmethod
+    @functools.lru_cache()
+    def render_graphiql() -> str:
+        """
+        Returns a string containing the html for the graphiql webpage. It also caches the
+        result using lru cache. This saves loading from disk each time it is invoked.
+        Returns:
+
+        """
+        result = render_graphiql_page()
+        return result
+
+    @staticmethod
+    def has_html_been_asked_for(headers: CaseInsensitiveMapping) -> bool:
+        """
+        Do the headers indicate that the invoker has requested html?
+        Args:
+            headers: A dictionary containing the headers in the request
+
+        Returns:
+            Whether html has been requested True for yes, False for no
+        """
+        try:
+            accept_headers = headers.get("accept")
+        except (KeyError, TypeError):
+            return False
+
+        if accept_headers is None:
+            return False
+
+        if "text/html" in accept_headers:
+            return True
+
+        if "*/*" in accept_headers:
+            return True
+
+        return False
+
+    @staticmethod
+    def method_is_post(method: str) -> bool:
+        """
+        If the get method has been called this is True
+        Args:
+            method: A request verb such as GET, POST, PUT etc.
+
+        Returns:
+            True if the method is post
+        """
+        if method == "POST":
+            return True
+
+        return False
```

Comment:
```suggestion
        return method == "POST"
```

Reply:
Not sure if we need dedicated functions for these.

Reply:
I have removed them